core/Option.php

```diff
@@ -190,12 +190,23 @@ protected function setValue($name, $value, $autoLoad = 0)
     {
         $autoLoad = (int)$autoLoad;
 
-        $sql  = 'INSERT INTO `' . Common::prefixTable('option') . '` (option_name, option_value, autoload) ' .
-                ' VALUES (?, ?, ?) ' .
-                ' ON DUPLICATE KEY UPDATE option_value = ?';
-        $bind = array($name, $value, $autoLoad, $value);
+        $sql  = 'UPDATE `' . Common::prefixTable('option') . '` SET option_value = ?, autoload = ? WHERE option_name = ?';
+        $bind = array($value, $autoLoad, $name);
 
-        Db::query($sql, $bind);
+        $result = Db::query($sql, $bind);
+
+        $rowsUpdated = Db::get()->rowCount($result);
+
+        if (! $rowsUpdated) {
+            try {
+                $sql  = 'INSERT INTO `' . Common::prefixTable('option') . '` (option_name, option_value, autoload) ' .
+                        'VALUES (?, ?, ?) ';
+                $bind = array($name, $value, $autoLoad);
+
+                Db::query($sql, $bind);
+            } catch (\Exception $e) {
```

Comment:
Any reason for catching exceptions here? We didn't used to do that before. Wondering if that might have any side effect always updating the `all` array, even if the insert failed

Reply:
`all`?

The proposed PR is no longer an atomic operation. So, the catch is for a potential race condition with concurrent requests to INSERT the same option name; i.e., only the first will succeed.

Reply:
Some lines below the 'static' array is updated `$this->all[$name] = $value;`. This will now be done, even if the second query fails to update.